Dockerfile

```diff
@@ -22,7 +22,16 @@ ADD . /go/src/github.com/intel/pmem-csi
 ENV GOPATH=/go
 ENV PKG_CONFIG_PATH=/usr/lib/pkgconfig/
 WORKDIR /go/src/github.com/intel/pmem-csi
-RUN make VERSION=${VERSION} OUTPUT_DIR=/go/bin
+ARG BIN_SUFFIX
+# Here we choose explicitly which binaries we want in the image and in
+# which flavor (production or testing). The actual binary name in the
+# image is going to be the same, to avoid unnecessary deployment
+# differences.
+RUN make VERSION=${VERSION} pmem-csi-driver${BIN_SUFFIX} pmem-vgm${BIN_SUFFIX} pmem-ns-init${BIN_SUFFIX} && \
+    mkdir -p /go/bin/ && \
```

Comment:
I would suggest using new make variable OUTPUT_DIR

Reply:
> I can't just copy all of OUTPUT_DIR like you did before because each binary must be renamed when building the test image (pmem-csi-driver-test -> pmem-csi-driver in the image). This is done in this stage.

Why can't `make` generate binaries without -test prefix, then we need not to do this extra step of renaming them. If someone wants to generate both(with and without -test) variants of binaries, could select with different `OUTPUT_DIR`s
> 
> I also think that is is better to explicitly list in the Dockerfile what ends up in the image, instead of relying on the Makefile and what it produces. The result is more predictable and easier to review this way.

If you want to choose what goes into the image, then I would prefer in the later stage, while preparing the final image.

Reply:
> Why can't make generate binaries without -test prefix

It's useful to build both locally.

> If you want to choose what goes into the image, then I would prefer in the later stage, while preparing the final image.

Why?

Reply:
I would prefer to define new `BUILD_VARIANT` build arg, instead of `BIN_SUFFIX`, and define below make targets:
```
build: $(CMDS)
build-test: $(addsuffix -test,$(CMDS))
build-all:
   make build
   make build-test OUTPUT_DIR=${OUTPUT_DIR}/test
```
In Dockerfile:
```
ARG BUILD_VARIANT
RUN make VERSION=${VERSION} build${BUILD_VARIANT} OUTPUT_DIR=/go/bin
```

This is so convenient to use, instead of renaming binaries.